File: SmartInventoryAPI/Models/Entities/Item.cs
namespace SmartInventoryAPI.Models.Entities;

public class Item
{
    public long ID { get; set; }
    public string? Item_Name { get; set; }
    public string? Item_Code { get; set; }
    public string? Description { get; set; }
    public string? Item_Category { get; set; }
    public string? Item_Brand { get; set; }
    public decimal Purchase_Cost { get; set; }
    public decimal Unit_Cost { get; set; }
    public bool Is_Active { get; set; } = true;
    public string? Unit_Of_Measure { get; set; }
    public string? Remark { get; set; }
    public DateTime Creation_Date { get; set; } = DateTime.UtcNow;
    public bool Is_Deleted { get; set; } = false;
    public string? Item_Image_URL { get; set; }
    public decimal Tax_Percentage { get; set; }
    public string? Tax_Type { get; set; }
    public string? Item_Type { get; set; }
}

File: SmartInventoryFrontend/src/types/item.ts
import { Entity } from './common'

export interface Item extends Entity {
  Item_Name: string
  Item_Code: string
  Description?: string
  Item_Category: string
  Item_Brand?: string
  Purchase_Cost: number
  Unit_Cost: number
  Is_Active: boolean
  Unit_Of_Measure: string
  Remark?: string
  Item_Image_Url?: string
  Tax_Percentage?: number
  Tax_Type?: string
  Item_Type?: string
}

export interface CreateItemRequest {
  Item_Name: string
  Item_Code: string
  Description?: string
  Item_Category: string
  Item_Brand?: string
  Purchase_Cost: number
  Unit_Cost: number
  Is_Active: boolean
  Unit_Of_Measure: string
  Remark?: string
  Item_Image_Url?: string
  Tax_Percentage?: number
  Tax_Type?: string
  Item_Type?: string
}

export interface UpdateItemRequest {
  Item_Name: string
  Item_Code: string
  Description?: string
  Item_Category: string
  Item_Brand?: string
  Purchase_Cost: number
  Unit_Cost: number
  Is_Active: boolean
  Unit_Of_Measure: string
  Remark?: string
  Item_Image_Url?: string
  Tax_Percentage?: number
  Tax_Type?: string
  Item_Type?: string
}

export interface ItemFilters {
  skip: number
  take: number
  searchQuery?: string
  category?: string
  isActive?: boolean
}

export interface ItemState {
  items: Item[]
  currentItem: Item | null
  loading: boolean
  error: string | null
  total: number
  skip: number
  take: number
  searchQuery: string | null
}
